module/VuFind/tests/unit-tests/src/VuFindTest/Role/PermissionProvider/UserTest.php

```diff
@@ -150,17 +151,27 @@ protected function getMockAuthorizationService()
     /**
      * Get a mock user object
      *
-     * @return \VuFind\Db\Row\User
+     * @return \VuFind\Db\Entity\UserEntityInterface
      */
-    protected function getMockUser(): \VuFind\Db\Row\User
+    protected function getMockUser()
     {
-        $user = $this->getMockBuilder(\VuFind\Db\Row\User::class)
-            ->disableOriginalConstructor()
-            ->getMock();
-        $user->method('__get')
-            ->will($this->returnValueMap($this->userValueMap[$this->testuser]));
-        $user->method('offsetGet')
-            ->will($this->returnValueMap($this->userValueMap[$this->testuser]));
+        $user = $this->createMock(UserEntityInterface::class);
+        if (!isset($this->userValueMap[$this->testuser])) {
+            return $user; // Return the mock as is if there's no test data
+        }
+
+        // Extract values correctly
+        $userData = [];
+        foreach ($this->userValueMap[$this->testuser] as $entry) {
+            if (isset($entry[0], $entry[1])) {
+                $userData[$entry[0]] = $entry[1];
+            }
+        }
+
+        // Dynamically mock getter methods
+        foreach ($userData as $property => $value) {
+            $user->method('get' . ucfirst($property))->willReturn($value);
+        }
```

Comment:
Some simplification suggestions:
  * I don't think we need to check whether the entry consists of two parts; since it's hard-coded test data, we can trust it to be structured correctly (or else the test will fail and we can fix it).
  * It might be more straightforward to build the mock in one step, rather than reformatting the data first and then looping a second time.
  * We can use null coalescing in the foreach loop to account for the missing user scenario without needing a separate if/return statement. 

What do you think of this version?

```suggestion
        // Dynamically mock getter methods
        foreach ($this->userValueMap[$this->testuser] ?? [] as $entry) {
            [$property, $value] = $entry;
            $user->method('get' . ucfirst($property))->willReturn($value);
        }
```